homeassistant/components/media_player/frontier_silicon.py

```diff
@@ -0,0 +1,254 @@
+        return SUPPORT_FRONTIER_SILICON
+
+    @property
+    def state(self):
+        """Return the state of the player."""
+        return self._state
+
+    # source
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return self._source_list
+
+    @property
+    def source(self):
+        """Name of the current input source."""
+        return self._source
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._media_image_url
+
+    def update(self):
+        """Get the latest date and update device state."""
+        fs_device = self.fs_device
+
+        if not self._name:
+            self._name = fs_device.friendly_name
+
+        if not self._source_list:
+            self._source_list = fs_device.mode_list
+
+        status = fs_device.play_status
+        self._state = {
+            'playing': STATE_PLAYING,
+            'paused': STATE_PAUSED,
+            'stopped': STATE_OFF,
+            'unknown': STATE_UNKNOWN,
+            None: STATE_OFF,
+        }.get(status, STATE_UNKNOWN)
+
+        info_name = fs_device.play_info_name
+        info_text = fs_device.play_info_text
+
+        self._title = '- '.join(filter(None, [info_name, info_text]))
+        self._artist = fs_device.play_info_artist
+        self._album_name = fs_device.play_info_album
+
+        self._source = fs_device.mode
+        self._mute = fs_device.mute
+        self._media_image_url = fs_device.play_info_graphics
+
+    # Management actions
+
+    # power control
+    def turn_on(self):
+        """Turn on the device."""
+        self.fs_device.power = True
+
+    def turn_off(self):
+        """Turn off the device."""
+        self.fs_device.power = False
+
+    def media_play(self):
+        """Send play command."""
+        self.fs_device.play()
+
+    def media_pause(self):
+        """Send pause command."""
+        self.fs_device.pause()
+
+    def media_play_pause(self):
+        """Send play/pause command."""
+        if 'playing' in self._state:
+            self.fs_device.pause()
+        else:
+            self.fs_device.play()
+
+    def media_stop(self):
+        """Send play/pause command."""
+        self.fs_device.pause()
+
+    def media_previous_track(self):
+        """Send previous track command (results in rewind)."""
+        self.fs_device.prev()
+
+    def media_next_track(self):
+        """Send next track command (results in fast-forward)."""
+        self.fs_device.next()
+
+    # mute
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self._mute
+
+    def mute_volume(self, mute):
+        """Send mute command."""
+        self.fs_device.mute = mute
+
+    # volume
+    def volume_up(self):
+        """Send volume up command."""
+        fs_device = self.fs_device
+        fs_device.volume = (fs_device.volume + 1)
```

Comment:
If you change this to `self.fs_device.volume += 1`, the property will only be accessed once without requiring the intermediate variable. Same below.